lti_store/models.py

```diff
@@ -0,0 +1,75 @@
+import uuid
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+
+def generate_client_id():
+    """
+    Generates a random UUID string.
+    """
+    return str(uuid.uuid4())
+
+
+class LTIVersion(models.TextChoices):
+    LTI_1P1 = "lti_1p1", _("LTI 1.1")
+    LTI_1P3 = "lti_1p3", _("LTI 1.3")
+
+
+class LTIConfiguration(models.Model):
+
+    name = models.CharField(max_length=80)
+    description = models.TextField(null=True, blank=True, default="")
+
+    version = models.CharField(
+        max_length=10, choices=LTIVersion.choices, default=LTIVersion.LTI_1P1
+    )
+
+    # This is where the configuration is stored
+    lti_config = models.JSONField(
+        null=False, blank=True, default=dict, help_text=_("LTI configuration data.")
+    )
+
+    # LTI 1.1 Related variables
+    lti_1p1_launch_url = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text=_("The URL of the external tool that initiates the launch."),
+    )
+    lti_1p1_client_key = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text=_("Client key provided by the LTI tool provider."),
+    )
+
+    lti_1p1_client_secret = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text=_("Client secret provided by the LTI tool provider."),
+    )
+
+    # LTI 1.3 Related variables
+    lti_1p3_internal_private_key = models.TextField(
+        blank=True,
+        help_text=_("Platform's generated Private key. Keep this value secret."),
+    )
+
+    lti_1p3_internal_private_key_id = models.CharField(
+        max_length=255,
+        blank=True,
+        help_text=_("Platform's generated Private key ID"),
+    )
+
+    lti_1p3_internal_public_jwk = models.TextField(
+        blank=True,
+        help_text=_("Platform's generated JWK keyset."),
+    )
+
+    lti_1p3_client_id = models.CharField(
+        max_length=255,
+        default=generate_client_id,
+        help_text=_("Client ID used by LTI tool"),
+    )
```

Comment:
We only support LTI 1.1 configurations, for now, you can remove these fields until we implement the new version.
```suggestion
```